python-scripts/eeprom-util.py

```diff
@@ -0,0 +1,198 @@
+    self.confounder = bytes(eepFile.read(8))      # encrypted confounder
+    self.hddKey = bytes(eepFile.read(16))         # encrypted HDD key
+    self.gameReg = bytes(eepFile.read(4))         # encrypted game region code
+
+    # identity variables
+    self.identChk = bytes(eepFile.read(4))        # identity checksum
+    self.serial = bytes(eepFile.read(12))         # serial number
+    self.mac = bytes(eepFile.read(6))             # MAC address
+    self.unknown0 = bytes(eepFile.read(2))        # TBD
+    self.onlineKey = bytes(eepFile.read(16))      # online key
+    self.videoStd = bytes(eepFile.read(4))        # NTSC / PAL
+    self.unknown1 = bytes(eepFile.read(4))        # TBD
+
+    # settings variables
+    self.settChk = bytes(eepFile.read(4))         # settings checksum
+    self.timeZoneBias = bytes(eepFile.read(4))    # time zone bias
+    self.timeZoneStdName = bytes(eepFile.read(4)) # standard timezone
+    self.timeZoneDltName = bytes(eepFile.read(4)) # daylight savings timezone
+    self.unknown2 = bytes(eepFile.read(8))        # TBD
+    self.timeZoneStdDate = bytes(eepFile.read(4)) # standard date / time
+    self.timeZoneDltDate = bytes(eepFile.read(4)) # daylight savings date / time
+    self.unknown3 = bytes(eepFile.read(8))        # TBD
+    self.timeZoneStdBias = bytes(eepFile.read(4)) # standard bias
+    self.timeZoneDltBias = bytes(eepFile.read(4)) # daylight savings bias
+    self.languageId = bytes(eepFile.read(4))      # language identifier
+    self.videoFlags = bytes(eepFile.read(4))      # video settings
+    self.audioFlags = bytes(eepFile.read(4))      # audio settings
+    self.gameParCont = bytes(eepFile.read(4))     # game parental contol settings
+    self.parContPass = bytes(eepFile.read(4))     # parental control password
+    self.dvdParCont = bytes(eepFile.read(4))      # DVD parental control settings
+    self.ip = bytes(eepFile.read(4))              # IP address
+    self.dns = bytes(eepFile.read(4))             # DNS server
+    self.gateway = bytes(eepFile.read(4))         # default gateway
+    self.subnet = bytes(eepFile.read(4))          # subnet
+    self.unknown4 = bytes(eepFile.read(4))        # TBD
+    self.dvdReg = bytes(eepFile.read(4))          # DVD region code
+    self.unknown5 = bytes(eepFile.read(64))       # TBD
+
+# Because of the confounder we have to split the RC4 decryption / encryption proccess
+# into two functions (eepKeyPrepCrypt / eepKeyCrypt). We also need a way to share state
+# between the the two functions (EepKeyObj). In a nutshell we fill and EepKeyObj method
+# object with the initial state by calling eepKeyPrepCrypt. Then we call eepKeyCrypt
+# on the confounder data. eepKeyCrypt will push state data back into the EepKeyObj method 
+# object (i / j) which is necessary when we call eepKeyCrypt on the next chunk of data
+# from the Eeprom method object (eeprom.hddKey + eeprom.gameReg). This is also why we
+# don't use any standard RC4 crypto libaries here.
```

Comment:
I'm not sure if I understand the issue. A standard RC4 implementation should also support this: https://wiki.openssl.org/index.php/Manual:Rc4(3)

In case it doesn't, please still mark / name your functions as being RC4ish and link to a source / explanation of the algorithm (Wikipedia might be fine) and pointing out the differences / details in usage.